pkg/stdcopy/stdcopy_test.go

```diff
@@ -49,6 +50,34 @@ func TestWrite(t *testing.T) {
 	}
 }
 
+type errWriter struct {
+	n   int
+	err error
+}
+
+func (f *errWriter) Write(buf []byte) (int, error) {
+	return f.n, f.err
+}
+
+func TestWriteWithWriterError(t *testing.T) {
+	expectedError := errors.New("expected")
+	writer := NewStdWriter(&errWriter{err: expectedError}, Stdout)
+	data := []byte("This won't get written, sigh")
+	_, err := writer.Write(data)
+	if err != expectedError {
```

Comment:
To be *complete* we should check the number of byte written (https://github.com/docker/docker/blob/master/pkg/stdcopy/stdcopy.go#L44) too I think :wink:.